app/models/profile.rb

```diff
@@ -40,6 +47,14 @@ def self.ransackable_attributes(_auth_object = nil)
     %w[name]
   end
 
+  def should_generate_new_friendly_id?
+    name_changed? || super
+  end
+
+  def name_or_anonymous
+    name || "anonymous"
+  end
```

Comment:
When creating a Profile from Avo, the param name is set as empty string "", therefore the slug looks just as a uuid:
```
e106520e-3908-4730-ba93-c31a15627966
```

Creating from console works as expected as name is nil.
I would suggest using `.presence` to ensure the behaviour

```
  def name_or_anonymous
    name.presence || "anonymous"
  end
```